plugins/rbac/src/hooks/useRoles.ts

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { useAsync, useAsyncRetry, useInterval } from 'react-use';
+
+import { useApi } from '@backstage/core-plugin-api';
+
+import { Role, RoleBasedPolicy } from '@janus-idp/backstage-plugin-rbac-common';
+
+import { rbacApiRef } from '../api/RBACBackendClient';
+import { RolesData } from '../types';
+import { getPermissions } from '../utils/rbac-utils';
+
+export const useRoles = () => {
+  const rbacApi = useApi(rbacApiRef);
+  const {
+    loading: rolesLoading,
+    value: roles,
+    retry,
+  } = useAsyncRetry(async () => {
+    return await rbacApi.getRoles();
+  });
+
+  const { loading: policiesLoading, value: policies } = useAsync(
+    async () => await rbacApi.getPolicies(),
+    [],
+  );
+
+  const data: RolesData[] = React.useMemo(
+    () =>
+      !roles || roles.length === 0
+        ? []
+        : roles.reduce(
+            (acc: any, role: Role) => [
+              ...acc,
+              {
+                id: role.name,
+                name: role.name,
+                description: '-',
+                members: role.memberReferences,
+                permissions: getPermissions(
+                  role.name,
+                  policies as RoleBasedPolicy[],
+                ),
+                modifiedBy: '-',
+                lastModified: '-',
+              },
+            ],
+            [],
+          ),
+    [roles, policies],
+  );
+  useInterval(() => retry(), 3000);
```

Comment:
shall we have it or provide user an option to configure it as optional and if not provided default to `3000`